frameworks/helloworld/integration/tests/test_upgrade.py

```diff
@@ -28,23 +23,41 @@ def teardown_module(module):
     uninstall()
 
 
-@pytest.mark.skip(reason="Currently fails...")
 @pytest.mark.sanity
-def test_upgrade():
+@pytest.mark.upgrade
+def test_upgrade_downgrade():
+    # Ensure both Universe and the test repo exist.
+    if len(shakedown.get_package_repos()['repositories']) != 2:
+        print('No test repo found.  Skipping test_upgrade_downgrade')
```

Comment:
Thanks, @nickbp The use case is our Framework Test Suite, which runs only packages from Universe. It doesn't add a test repo like the PR jobs. I will add a comment.